indexing-service/src/main/java/io/druid/indexing/overlord/autoscaling/PendingTaskBasedWorkerProvisioningStrategy.java

```diff
@@ -53,56 +52,74 @@
 {
-  private static final EmittingLogger log = new EmittingLogger(PendingTaskBasedWorkerResourceManagementStrategy.class);
+  private static final EmittingLogger log = new EmittingLogger(PendingTaskBasedWorkerProvisioningStrategy.class);
 
-  private final PendingTaskBasedWorkerResourceManagementConfig config;
+  private final PendingTaskBasedWorkerProvisioningConfig config;
   private final Supplier<WorkerBehaviorConfig> workerConfigRef;
-  private final ScalingStats scalingStats;
-
-  private final Object lock = new Object();
-  private final Set<String> currentlyProvisioning = Sets.newHashSet();
-  private final Set<String> currentlyTerminating = Sets.newHashSet();
-
-  private DateTime lastProvisionTime = new DateTime();
-  private DateTime lastTerminateTime = new DateTime();
 
   @Inject
-  public PendingTaskBasedWorkerResourceManagementStrategy(
-      PendingTaskBasedWorkerResourceManagementConfig config,
+  public PendingTaskBasedWorkerProvisioningStrategy(
+      PendingTaskBasedWorkerProvisioningConfig config,
       Supplier<WorkerBehaviorConfig> workerConfigRef,
-      ResourceManagementSchedulerConfig resourceManagementSchedulerConfig,
-      ScheduledExecutorFactory factory
+      ProvisioningSchedulerConfig provisioningSchedulerConfig
   )
   {
     this(
         config,
         workerConfigRef,
-        resourceManagementSchedulerConfig,
-        factory.create(1, "PendingTaskBasedResourceManagement-manager--%d")
+        provisioningSchedulerConfig,
+        new Supplier<ScheduledExecutorService>()
+        {
+          @Override
+          public ScheduledExecutorService get()
+          {
+            return ScheduledExecutors.fixed(1, "PendingTaskBasedResourceManagement-manager--%d");
+          }
+        }
     );
   }
 
-  public PendingTaskBasedWorkerResourceManagementStrategy(
-      PendingTaskBasedWorkerResourceManagementConfig config,
+  public PendingTaskBasedWorkerProvisioningStrategy(
+      PendingTaskBasedWorkerProvisioningConfig config,
       Supplier<WorkerBehaviorConfig> workerConfigRef,
-      ResourceManagementSchedulerConfig resourceManagementSchedulerConfig,
-      ScheduledExecutorService exec
+      ProvisioningSchedulerConfig provisioningSchedulerConfig,
+      Supplier<ScheduledExecutorService> execFactory
   )
   {
-    super(resourceManagementSchedulerConfig, exec);
+    super(provisioningSchedulerConfig, execFactory);
     this.config = config;
     this.workerConfigRef = workerConfigRef;
-    this.scalingStats = new ScalingStats(config.getNumEventsToTrack());
   }
 
   @Override
-  public boolean doProvision(WorkerTaskRunner runner)
+  public Provisioner makeProvisioner(WorkerTaskRunner runner)
   {
-    Collection<Task> pendingTasks = runner.getPendingTaskPayloads();
-    Collection<ImmutableWorkerInfo> workers = runner.getWorkers();
-    synchronized (lock) {
+    return new PendingProvisioner(runner);
+  }
+
+  private class PendingProvisioner implements Provisioner
```

Comment:
Explored this but didn't find anything that could be extracted as a method with a sensible name and "abstraction", so didn't extract